### ISSUE TITLE
`using assembly` is processed at compile time rather than parse time

### ISSUE THREAD

**Issue by SeeminglyScience** [PowerShell/PowerShell#18482]
### Steps to reproduce

I know this issue probably exists in some form but I want to get it recorded quickly so I can add it to the agenda for the engine WG. This mostly came up due to what I suspect is the docs describing how we'd like it work, vs how it currently works.

More specifically, [the docs state](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_using?view=powershell-7.2#assembly-syntax):

(emphasis is mine)

> Loading an assembly preloads .NET types from that assembly into a script **at parse time**. This allows you to **create new PowerShell classes that use types from the preloaded assembly**.

```powershell
using assembly SomeAssembly.dll

class MyClass : SomeTypeFromSomeAssembly {
}
```

### Expected behavior

Either we make it work (one day) or update the docs

### Actual behavior

```console
ParserError:
Line |
   3 |  class MyClass : SomeTypeFromSomeAssembly {
     |                  ~~~~~~~~~~~~~~~~~~~~~~~~
     | Unable to find type [SomeTypeFromSomeAssembly].
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.0-rc.1
PSEdition                      Core
GitCommitId                    7.3.0-rc.1
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by 237dmitry** [comment]
```powershell
ParserError: /data/0.ps1:1
Line |
   1 |  using assembly SomeAssembly.dll
     |                 ~~~~~~~~~~~~~~~~
     | Cannot load assembly 'SomeAssembly.dll'

```

**Comment by SeeminglyScience** [comment]
Yeah the example is arbitrary, I just wanted to get something quick put up so it could be discussed. The parser attempts to verify that the assembly exists, but does not actually try to load it at parse time.

**Comment by mklement0** [comment]
See also:

* https://github.com/PowerShell/PowerShell/issues/3641#issuecomment-297151112

**Comment by mklement0** [comment]
A more detailed technical discussion of a potential implementation starts here:

* https://github.com/PowerShell/PowerShell/issues/2074#issuecomment-428299006

**Comment by SeeminglyScience** [comment]
The help document in question has been updated, marking as resolved.
